docs/proposals/timeouts.md

```diff
@@ -0,0 +1,119 @@
+    development. Comparing this with the problem of failed DeployItems in productive systems under heavy load, seems to
+    be a bad deal. Furthermore, for the development scenario, it is already possible to find out that a DeployItem
+    is not picked up, by its empty status and such DeployItems could be stopped by the interrupt annotation. 
+    Therefore, the removal of this timeout or setting it on a high value might be a good idea. 
+  - If we want to be more defencive, we could add a pickup timeout to the DeployItem specification itself. Then the 
+    customer could decide by itself if he wants this feature e.g. in his development scenario.
+
+- Reduce number of processing timeout settings
+  - Motivation: It is less complicated for a user to only set one timeout for the deployment or deletion of a DeployItem.
+    Even finding here the right value is already complicated but to take all the other current settings into account
+    is nearly impossible.
+  - Only one processing timeout on deploy item level which determines how long the deployment is allowed to require.
+  - Only one delete timeout on deploy item level which determines how long the deletion is allowed to require.
+    (if not set, use some default or use same as for progressing timeout)
+  - Timeout controller:
+    - Exclude the manifest and helm deployer from the timeout controller
+    - When we have also found a similar solution for the container deployer we could perhaps completely remove the central
+      processing timeout controller? 
+  - Deployment/Upgrade/Deletion
+    - Check the timeout between the main steps like templating, deployment, readiness checks, export generation. A
+      particular step gets an allowed duration computed by (timeout - required duration of steps before). When the timout 
+      is exceeded the processing is interrupted and the DeployItem set on failed. 
+      - Running requests like helm 3 deployments or manifest deployments are canceled if they require too much time
+      - The helm 3 timeout is recomputed with respect to the remaining duration - no matter if it is set or not.
+    - Deletions are handled the same way. 
+
+## Open Question
+
+- When does the processing timeout start? When a DeployItem is triggered of when the deployer starts working on it?
+  - Alternative one has the advantage, that a system under load, where it might need some time until a deployer could
+    pick up a DeployItem, is not running out of time. It seems also to be much easier for a customer to know how long a 
+    deployment for a DeployItems needs, compared to how long a deployer requires to pick up a DeployItem.
+  
+- In case of an error and an immediate retry of the reconciliation of the DeployItem, should the required time of the 
+  erroneous reconciliations from before be taken into account and removed for the allowed processing time? 
+  Probably yes as it prevents infinite loops.
```

Comment:
Hm, but the controller-runtime applies some exponential backoff logic for repeatedly failed reconciliations. Imagine a DI which talks to a server and that server is down for five minutes. When the server is back up and the repeatedly failed DI is reconciled again, it immediately runs into a progressing timeout, because the past five minutes of trying count into the timeout duration? Doesn't sound like a good idea to me ...

I don't think that repeated failures of a DI are a problem, because the DI will be in Error state for most of the time, so it's easy to recognize. But maybe I'm missing something here.